src/lib/utils/thorchain/memo.ts

```diff
@@ -0,0 +1,136 @@
+import { bn } from 'lib/bignumber/bignumber'
+
+import { THORCHAIN_AFFILIATE_NAME } from './constants'
+
+const assertIsValidPool = (pool: string, memo: string) => {
+  if (!pool) throw new Error(`pool is required in memo: ${memo}`)
+}
+
+const assertIsValidAsset = (asset: string, memo: string) => {
```

Comment:
Same with all of these helper functions.

Reply:
570726bea1